src/bpftrace.cpp

```diff
@@ -1758,11 +1758,30 @@ std::string BPFtrace::get_stack(uint64_t stackidpid, bool ustack, StackType stac
                             stack_trace.data());
   if (err)
   {
-    // ignore EFAULT errors: eg, kstack used but no kernel stack
-    if (stackid != -EFAULT)
+    if (stackid == -EFAULT && stack_type.type == StackType::ukstack::kstack)
```

Comment:
We've actually got the ustack / kstack information already available as an argument to this function: `ustack`. I'm not opposed to a potential refactor, but there's some unnecessary duplication with this method and I'm not sure how to get rid of it nicely